3rdParty/iresearch/core/utils/object_pool.hpp

```diff
@@ -625,7 +625,7 @@ class unbounded_object_pool : public unbounded_object_pool_base<T> {
     node* head = nullptr;
 
     // reset all cached instances
-    while (head = this->free_objects_.pop()) {
```

Comment:
Is this necessary? Additional brackets seems redundant

Reply:
Compilers warn about assignments used as conditionals, for clang and gcc it's `-Wparentheses`, to catch accidental usage of `=` vs `==`. This warning can be suppressed by adding additional parentheses; the convention seemingly being that this means "I really meant an assignment".